benchmarks/benchmarks/long/esmf_regridder.py

```diff
@@ -0,0 +1,176 @@
+"""Slower benchmarks for :mod:`esmf_regrid.esmf_regridder`."""
+
+from pathlib import Path
+
+import numpy as np
+import dask.array as da
+import iris
+from iris.cube import Cube
+
+from benchmarks import disable_repeat_between_setup
+from esmf_regrid.experimental.unstructured_scheme import (
+    GridToMeshESMFRegridder,
+    MeshToGridESMFRegridder,
+)
+from esmf_regrid.schemes import ESMFAreaWeightedRegridder
+from esmf_regrid.tests.unit.schemes.test__cube_to_GridInfo import _grid_cube
+
+
+class PrepareScalabilityGridToGrid:
+    params = [50, 100, 500, 1000, 2000]
+    height = 100
+    regridder = ESMFAreaWeightedRegridder
+
+    def src_cube(self, n):
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        src = _grid_cube(n, n, lon_bounds, lat_bounds)
+        return src
+
+    def tgt_cube(self, n):
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        grid = _grid_cube(n + 1, n + 1, lon_bounds, lat_bounds)
+        return grid
+
+    def setup(self, n):
+        self.src = self.src_cube(n)
+        self.tgt = self.tgt_cube(n)
+
+    def time_prepare(self, n):
+        _ = self.regridder(self.src, self.tgt)
+
+
+class PrepareScalabilityMeshToGrid(PrepareScalabilityGridToGrid):
+    regridder = MeshToGridESMFRegridder
+
+    def src_cube(self, n):
+        from esmf_regrid.tests.unit.experimental.unstructured_scheme.test__mesh_to_MeshInfo import (
+            _gridlike_mesh,
+        )
+
+        src = Cube(np.ones([n * n]))
+        mesh = _gridlike_mesh(n, n)
+        mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+        src.add_aux_coord(mesh_coord_x, 0)
+        src.add_aux_coord(mesh_coord_y, 0)
+        return src
+
+
+class PrepareScalabilityGridToMesh(PrepareScalabilityGridToGrid):
+    regridder = GridToMeshESMFRegridder
+
+    def tgt_cube(self, n):
+        from esmf_regrid.tests.unit.experimental.unstructured_scheme.test__mesh_to_MeshInfo import (
+            _gridlike_mesh,
+        )
+
+        tgt = Cube(np.ones([(n + 1) * (n + 1)]))
+        mesh = _gridlike_mesh(n + 1, n + 1)
+        mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+        tgt.add_aux_coord(mesh_coord_x, 0)
+        tgt.add_aux_coord(mesh_coord_y, 0)
+        return tgt
+
+
+@disable_repeat_between_setup
+class PerformScalabilityGridToGrid:
+    params = [10, 100, 1000, 10000]
```

Comment:
The problem with scaling by `grid_size` is that large grids become dificult to set up. Setting up large grids will exhaust the performance and cause failures before we have a chance to see the effect of large data on the scaling of performance. It is also worth noting that we expect data with very large non-grid dimensions as a common use case. Scaling of performance based on grid size is also worth considering and it may be worth adding later.

Reply:
We've agreed offline to revisit extra parameterisation later, even if we don't have a machine that can handle it.